versioned/storage/cache/src/main/java/org/projectnessie/versioned/storage/cache/CaffeineCacheBackend.java

```diff
@@ -311,6 +338,26 @@ public int hashCode() {
     public String toString() {
       return "{" + repositoryId + ", " + id + '}';
     }
+
+    Obj getObj() {
+      Obj obj = (Obj) this.object.get();
+      if (obj == null) {
+        obj = ProtoSerialization.deserializeObj(id, 0L, this.serialized, null);
+        // re-create the soft reference - but don't care about JMM side effects
+        this.object = new SoftReference<>(obj);
```

Comment:
Should we perhaps adjust defaults in `CacheSizing` to allow for more "free" heap for the soft-referenced objects?

Reply:
From 70% to 60%?

Reply:
SGTM